### PR TITLE
Waiting room

### PR DESCRIPTION
**Pull request by kunalpunjrath**
All waiting room functionalities covered. Just need to integrate/ wrap with sockets.io events.

### CODE REVIEW

File: server/room.controller.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  if (room.isAvailable) {
+    //socket.broadcast.emit('broadcast(): player ' + playerId + ' joined room with topic' + topicId);
+    room.addPlayer(playerId);
+    //socket.emit("playerConnected", {playerId, topicId});
+  }
+};
+
+const removeRoom = function(topicId) {
+  const roomIndex = waitingRooms.findIndex((r) => {
+    return r.topicId === topicId;
```
Comment (anirvann) [inline server/room.controller.js:37]: r.getRoom() ?? And isnt the property name _topic?

**Review by shereej — COMMENTED** [review]
@kunalpunjrath what I understand is you still need to write socket emit for your events? and may I know why you removed the socket emit you have already had! :(

File: server/room.controller.js (hunk 1 above)
Comment (kunalpunjrath) [reply server/room.controller.js:37]: there is a get topicId in room.js

**Comment by kunalpunjrath** [comment]
Socket emits are all there but commented. I need to remove the commented portions and test from the client side.

**Pull request merged**